build/openresty/patches/ngx_lua-0.10.25_01-dyn_upstream_keepalive.patch

```diff
@@ -670,13 +685,26 @@ index af4da73..407c115 100644
 +            }
 +        }
 +
-+        return;
-+    }
-+
+         return;
+     }
+ 
+-    /* fallback */
 +    bp->original_free_peer(pc, bp->data, state);
 +}
 +
 +
++static void
++ngx_http_lua_balancer_notify_peer(ngx_peer_connection_t *pc, void *data,
++    ngx_uint_t state)
++{
++    ngx_http_lua_balancer_peer_data_t          *bp = data;
++
++    if (ngx_http_lua_balancer_keepalive_is_enabled(bp)){
```

Comment:
This check doesn't seem to be necessary. If `u->peer.cached` is true, balancer keepalive must have been enabled.

Reply:
right, fix it later